heuristic/repair_operators/greedy_insert.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+from numpy.random import RandomState
+import heapq
+
+
+from heuristic.classes import Item, Route, Solution, Stacks
+from heuristic.constants import DEPOT
+
+
+def greedy_insert(current: Solution, rnd_state: RandomState) -> Solution:
+    """
+    TODO.
+    """
+
+    while len(current.unassigned) != 0:
+        customer = current.unassigned.pop()
+        insertion_heap = []
+
+        for idx_route, route in enumerate(current.routes):
+            idx = route.opt_insert(current.problem, customer)
+            cost = route.insert_cost(current.problem, idx, customer)
+
+            if route.is_feasible(current.problem, idx):
+                heapq.heappush(insertion_heap, (cost, idx, idx_route))
+
+        # Costs new route
+        cost_new_route = ([current.problem.distances[DEPOT + 1, customer + 1] +
+                           current.problem.distances[customer + 1, DEPOT + 1]])
+        heapq.heappush(insertion_heap, (cost_new_route, 0, len(current.routes)))
```

Comment:
Why does this need to be on the heap? Can't you compare this to the routes on the heap to determine if it's worthwhile? That'd allow you to store routes on the heap, which is just that much easier.